README.md
# DBFZ_Team_Random V 1.3.2
Built this for my free time, me and a good friend of mine like to roll random teams in Dragonball FighterZ as you cannot random while playing online, only while local.

# built with:
  * Javascript
  * CSS3
  * HTML5

## Extra:
i don't know why buy i am strangely proud of the little work i did on this, since its a little nieche and not a lot of honest work it was just the sheer fact that i came up with an idea of it while trying to do random teams against a freind of mine, and that alond drove me far enough to be able to build this out, host it oh GH, and make sure it was working more that just nth % amount of times.


ChangeLog.md
# version 1.3.2 dec 26 2023

## **Notes**:
nothing to major in this update it was largely a refactor of what already existed to make it more fluid and purpouseful.

### **Changed**:
- some CSS around heights of the current random team and the last team have changed to make the page more consistent.
- refactored most of the logic to be more compact and understood.
- tried to change the CSS around other device sizes so little to no visual oddities occur.

# version 1.3.1 dec 10 2023
## **Notes**:
Found some not so great renderings when using the site recently and wanted to fix it.
### **Changed**:
- some of the css style options for more consistent page renders at all sizes.
- fixed some other oddities that would happen on rendering a new team, mainly heights suddenly popping in when an image was rendered

# version 1.3 apr 30 2023

## **Notes**:
Wanted to add a few features to the site that would help out those who may not have all ther characters in the game, as well as a way to keep track of the assists that the last team was using.

### **Added**:
- **Assists** have been added the last roll section **IF** you have assists set to be rolled.
- **DLC lists** have been added to the page to allow those who may not have access to specific characters. this may be expanded later into a full blown list, but i want to keep things digustable.

### **Changed**:
- **Spacing** don't want to boar you with the technical details but some work has been done to keep the app more centered in all screen sizes.
- **Backgrounds** will be a noticebly different in some places to keep the ui from being too samey.

### **Changed**:

# version 1.2 apr 19 2023

## **Notes**:
wanted to add a characer that was missing **Android 21 (Lab Coat)** is now in the randomizer, and with her inclusion required updates to the `UX/UI`.

### **Added**: 
- **Android 21 (Lab Coat)**: has been added to the randomization pool
- **Mobile friendly design-ish**: many elements had to be improved and made the site more Mobile friendly
    - this process will be ongoing even if looking finished.

### **Changed**:
- **images** will look smaller as their size has been adjusted for a consistent look.
- **text** has been changed to keep with a consistent look and sizing
- **background** spacing changed to have a "fuller" background

# version 1.1 - apr 9 2023

## **Notes**:
wanted to add a last team to help capture what a double roll, or to keep in mind what your last roll was. at this time it does not capture what assist each character was using.

### **Added**:
- **last roll**: a section that will capture the **last team** you randomized.
  - it currently **doesn't** capture the **assist rolls.**

### **Changed**:
- updated some of the alignemnt of text, selectors, and backgrounds for a more conistent experience.  

# version 1.0 - jan 25 2022

## Randomizer Initial realease

### **Added**:
- Randomizing characters
- Randomizing Assists


assets/javascript/index.js
// team positons
const point = document.getElementById('Point');
const mid = document.getElementById('Mid');
const anchor = document.getElementById('Anchor');

// interation layer
const btn = document.getElementById('btn');
const assistCheck = document.getElementById('assist');
    // seasons
const seasonOne = document.getElementById('season-1')
const seasonOnelist = document.getElementById('season-1-list')
    //
const seasonTwo = document.getElementById('season-2')
const seasonTwolist = document.getElementById('season-2-list')
    //
const seasonThree = document.getElementById('season-3')
const seasonThreelist = document.getElementById('season-3-list')
    //
const a21check = document.getElementById('android21')

// last roll layer
const lastRoll = document.getElementById('lastRoll');
const hidden = document.getElementById('hidden')

// characters from roaster.js
const roaster = Characters

let lastTeam = []

let blacklist = []

const team = [
    {
        name: "",
        num: 0,
        assist: '',
        position: point
    },
    {
        name: "",
        num: 0,
        assist: '',
        position: mid
    },
    {
        name: "",
        num: 0,
        assist: '',
        position: anchor
    }
]

const assistsArray = ['A', 'B', 'C']

function checkerHandler(children, checked) {
    for (const child of children) {
        child.children[0].checked = checked
        if (!checked) {
            if (!blacklist.includes(child.children[0].value)) {
                blacklist.push(child.children[0].value)
            }
            continue
        }
        const newlist = blacklist.filter(list => {
            if (child.children[0].value === list) {
                return
            }
            return list
        })
        blacklist = newlist
    }
}


function checkboxHandler(e) {
    const name = e.target.name;
    const checked = e.target.checked;
    switch (name) {
        case 'season-1':
            checkerHandler(seasonOnelist.children, checked)
            break
        case 'season-2':
            checkerHandler(seasonTwolist.children, checked)
            break
        case 'season-3':
            checkerHandler(seasonThreelist.children, checked)
            break
        default:
            if (checked != undefined) {
                checkerHandler([{ children: [e.target] }], checked)
            }
            break
    }
}

function printLastTeam() {
    lastRoll.replaceChildren()
    for (const player of lastTeam) {
        // =Container=
        var characterContainer = document.createElement("div");
        characterContainer.classList = "oldChraCont"
        // =Image=
        var lastCharacter = document.createElement("img");
        lastCharacter.src = "./assets/images/" + player.num + ".png";
        characterContainer.appendChild(lastCharacter);
        // =Assist=
        if (assistCheck.checked) {
            var assistContainer = document.createElement("h4");
            assistContainer.innerText = player.assist
            characterContainer.appendChild(assistContainer);
        }
        // =Append=
        lastRoll.appendChild(characterContainer);
    }
    hidden.classList = "oldTeam"
}

function printTeam() {
    // this is where everything gets rendered, starting with point characters.
    for (const player of team) {
        var charTitle = document.createElement("h1");
        charTitle.classList = "charName"
        charTitle.innerText = player.name;

        player.position.replaceChildren(charTitle)

        var charPort = document.createElement("img");
        charPort.src = `./assets/images/${player.num}.png`

        player.position.appendChild(charPort)

        if (assistCheck.checked) {
            var assistHtml = document.createElement("h3");
            const assist = player.assist;
            assistHtml.textContent = `${assist} Assist`

            player.position.appendChild(assistHtml)
        }
    }
}

function newTeam() {
    let newRoaster = roaster.filter(ele => {
        const checker = blacklist.includes(`${ele.num}`);
        if (checker) {
            return
        }
        return ele
    })
    for (const position of team) {
        const n = Math.floor(Math.random() * newRoaster.length)

        position.name = newRoaster[n].name;
        position.num = newRoaster[n].num;
        position.assist = assistsArray[Math.floor(Math.random() * 6 / 2)]

        const evenNewerRoaster = newRoaster.filter(character => character.num != newRoaster[n].num)
        newRoaster = evenNewerRoaster
    }
}

const ButtonHandler = (e) => {
    // this is for saving the last team to refrence in a component
    if (team[0].name) {
        const currentTeam = team
        lastTeam = currentTeam
        printLastTeam()
    }
    newTeam();
    printTeam();
}

btn.addEventListener("click", ButtonHandler)
seasonOne.addEventListener("click", checkboxHandler)
seasonTwo.addEventListener("click", checkboxHandler)
seasonThree.addEventListener("click", checkboxHandler)
seasonOnelist.addEventListener("click", checkboxHandler)
seasonTwolist.addEventListener("click", checkboxHandler)
seasonThreelist.addEventListener("click", checkboxHandler)
a21check.addEventListener("click", checkboxHandler)

roaster.js
const Characters = [
        {
            "name": "Goku (SSJ)",
            "num": 0
        },
        {
            "name": "Vegeta (SSJ)",
            "num": 1
        },
        {
            "name": "Piccolo",
            "num": 2
        },
        {
            "name": "Teen gohan",
            "num": 3
        },
        {
            "name": "Frieza",
            "num": 4
        },
        {
            "name": "Ginyu",
            "num": 5
        },
        {
            "name": "Trunks",
            "num": 6
        },
        {
            "name": "Cell",
            "num": 7
        },
        {
            "name": "Android 18",
            "num": 8
        },
        {
            "name": "Gotenks",
            "num": 9
        },
        {
            "name": "Krillin",
            "num": 10
        },
        {
            "name": "Kid Buu",
            "num": 11
        },
        {
            "name": "Majin Buu",
            "num": 12
        },
        {
            "name": "Nappa",
            "num": 13
        },
        {
            "name": "Android 16",
            "num": 14
        },
        {
            "name": "Yamcha",
            "num": 15
        },
        {
            "name": "Tien",
            "num": 16
        },
        {
            "name": "Adult gohan",
            "num": 17
        },
        {
            "name": "Hit",
            "num": 18
        },
        {
            "name": "Goku (SSGSS)",
            "num": 19
        },
        {
            "name": "Vegeta (SSGSS)",
            "num": 20
        },
        {
            "name": "Beerus",
            "num": 21
        },
        {
            "name": "Goku black",
            "num": 22
        },
        {
            "name": "Android 21",
            "num": 23
        },
        {
            "name": "Goku",
            "num": 24
        },
        {
            "name": "Vegeta",
            "num": 25
        },
        {
            "name": "Broly (DBZ)",
            "num": 26
        },
        {
            "name": "Zamasu",
            "num": 27
        },
        {
            "name": "Bardock",
            "num": 28
        },
        {
            "name": "Vegetio",
            "num": 29
        },
        {
            "name": "Android 17",
            "num": 30
        },
        {
            "name": "Cooler",
            "num": 31
        },
        {
            "name": "Jiren",
            "num": 32
        },
        {
            "name": "Videl",
            "num": 33
        },
        {
            "name": "Goku (GT)",
            "num": 34
        },
        {
            "name": "Junemba",
            "num": 35
        },
        {
            "name": "Gogeta (SSGSS)",
            "num": 36
        },
        {
            "name": "Broly (DBS)",
            "num": 37
        },
        {
            "name": "Kefla",
            "num": 38
        },
        {
            "name": "Goku (ui)",
            "num": 39
        },
        {
            "name": "Master Roshi",
            "num": 40
        },
        {
            "name": "Super Baby 2",
            "num": 41
        },
        {
            "name": "Gogeta (SS4)",
            "num": 42
        },
        {
            "name": "Android 21 (Lab Coat)",
            "num": 43
        }
    ]